qf_lib/data_providers/sp_global/sp_data_provider.py

```diff
@@ -0,0 +1,359 @@
+from qf_lib.common.tickers.tickers import Ticker, SPTicker
+from qf_lib.common.utils.miscellaneous.to_list_conversion import convert_to_list
+from qf_lib.containers.dataframe.qf_dataframe import QFDataFrame
+from qf_lib.containers.futures.future_tickers.future_ticker import FutureTicker
+from qf_lib.containers.series.qf_series import QFSeries
+from qf_lib.data_providers.abstract_price_data_provider import AbstractPriceDataProvider
+from qf_lib.data_providers.helpers import tickers_dict_to_data_array, normalize_data_array
+from sqlalchemy import and_, or_, func
+from sqlalchemy.orm import Session, aliased
+
+
+class SPDataProvider(AbstractPriceDataProvider, SPDAO):
+    def __init__(self, db_connection_provider: DBConnectionProvider, use_adjusted_prices: bool = True,
+                 exchange_rate_snap_id: int = 6):
+        """
+        Class responsible for fetching the market data from S&P Global. Requires the Market Data package.
+
+        Parameters
+        ----------
+        use_adjusted_prices: bool
+            Adjust the prices for corporate actions (mergers, spin-offs). The adjustment is always applied as
+            of end_datetime. If a corporate action occurred afterwards, it is not taken into account when computing
+            prices adjustment. By default True.
+        exchange_rate_snap_id: int
+            ID corresponding to the snapshot of the exchange rates used. Available values are:
+                1	- Sydney Midday
+                2	- Tokyo Midday
+                3	- Sydney Close
+                4	- Tokyo Close
+                5	- London Midday
+                6	- London Close
+                7	- New York Midday
+                8	- New York Close
+            Default value is set to 6 (London Close).
+        """
+        AbstractPriceDataProvider.__init__(self)
+        SPDAO.__init__(self, db_connection_provider)
+
+        self.use_adjusted_prices = use_adjusted_prices
+        if exchange_rate_snap_id not in range(1, 9):
+            raise ValueError(f"Incorrect Snapshot ID {exchange_rate_snap_id}. Possible values are integers between "
+                             f"1 and 8. Consult the documentation to see what each snapshot ID corresponds to.")
+        self.exchange_rate_snap_id = exchange_rate_snap_id
+
+    @property
+    def supported_fields(self) -> list[SPField]:
+        return [SPField.AskPrice, SPField.BidPrice, SPField.LowPrice, SPField.HighPrice, SPField.OpenPrice,
+                SPField.ClosePrice, SPField.Volume, SPField.DivYield]
+
+    def price_field_to_str_map(self) -> Dict[PriceField, SPField]:
+        return {
+            PriceField.Open: SPField.OpenPrice,
+            PriceField.High: SPField.HighPrice,
+            PriceField.Low: SPField.LowPrice,
+            PriceField.Close: SPField.ClosePrice,
+            PriceField.Volume: SPField.Volume
+        }
+
+    def get_history(self, tickers: Union[SPTicker, Sequence[SPTicker]], fields: Union[SPField, Sequence[SPField]],
```

Comment:
Will this ever return something else than a `QFDataArray`?

Reply:
Yes, whenever we have a single date, single ticker, single field the normalize_data_array will squeeze the array to am object with less dimensions